pkg/minikube/node/config_test.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"testing"
+)
+
+func Test_maskProxyPassword(t *testing.T) {
+	type dockerOptTest struct {
+		input  string
+		output string
+	}
+	var tests = []dockerOptTest{
+		{
+			input:  "cats",
+			output: "cats",
+		},
+		{
+			input:  "myDockerOption=value",
+			output: "myDockerOption=value",
+		},
+		{
+			input:  "http_proxy=http://myproxy.company.com",
+			output: "HTTP_PROXY=http://myproxy.company.com",
+		},
+		{
+			input:  "https_proxy=http://jdoe@myproxy.company.com:8080",
+			output: "HTTPS_PROXY=http://jdoe@myproxy.company.com:8080",
+		},
+		{
+			input:  "https_proxy=https://mary:am$uT8zB(rP@myproxy.company.com:8080",
```

Comment:
lets make this word obvious it is a fake password
such as
mary@iam$Fake!password

also lets make more example of password with other types of characters such as  ' %&* (things that could break our regex)

Reply:
Improved the regex robustness and the tests. I can't remember torturing regexes that much since I used to parse sendmail logs on SunOS 4.. 😊
Anyway, since this is my first PR and I'm tankful for the feedback. I hope it's better now.